hibernate-core/src/main/java/org/hibernate/query/criteria/internal/path/PluralAttributePath.java

```diff
@@ -44,39 +46,75 @@ private CollectionPersister resolvePersister(CriteriaBuilderImpl criteriaBuilder
 	}
 
```

Comment:
The following function has multiple returns and the flow is unclear. It looks like a lot of changes, but the changes are mostly due to changing to a switch statement.